src/pybacked.rs

```diff
@@ -324,6 +390,34 @@ mod test {
         });
     }
 
+    #[test]
+    fn py_backed_bytes_into_py() {
+        Python::with_gil(|py| {
+            let orig_bytes = PyBytes::new_bound(py, b"abcde");
+            let py_backed_bytes = PyBackedBytes::from(orig_bytes.clone());
+            assert!(py_backed_bytes.to_object(py).is(&orig_bytes));
+            assert!(py_backed_bytes.into_py(py).is(&orig_bytes));
+        });
+    }
+
+    #[test]
+    fn rust_backed_bytes_into_py() {
+        Python::with_gil(|py| {
+            let orig_bytes = PyByteArray::new_bound(py, b"abcde");
+            let rust_backed_bytes = PyBackedBytes::from(orig_bytes);
+            assert!(matches!(
+                rust_backed_bytes.storage,
+                PyBackedBytesStorage::Rust(_)
+            ));
+            let to_object = rust_backed_bytes.to_object(py).into_bound(py);
+            assert!(&to_object.is_exact_instance_of::<PyBytes>());
+            assert_eq!(&to_object.extract::<PyBackedBytes>().unwrap(), b"abcde");
+            let into_py = rust_backed_bytes.into_py(py);
+            assert!(&to_object.is_exact_instance_of::<PyBytes>());
+            assert_eq!(&into_py.extract::<PyBackedBytes>(py).unwrap(), b"abcde");
```

Comment:
I think the test didn't fail because this `.extract()` will presumable be creating a `PyBackBytesStorage::Python` form from the python `bytes`.

Reply:
Another oversight on my part... line `416` was re-checking the type of the `to_object` variable instead of `into_py` as intended. Fixing that was able to catch the bug. All should be good now